src/api.cpp

```diff
@@ -0,0 +1,301 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2021 YADRO
+
+#include <errno.h>
+#include <stdio.h>
+
+#include <libobmcsession/obmcsession.hpp>
+#include <manager.hpp>
+#include <session.hpp>
+
+using namespace obmc::session;
+static SessionManagerPtr managerPtr;
+static std::unique_ptr<sdbusplus::bus::bus> dbus;
+
+int obmcsesManagerInit(sd_bus* bus, const char* slug, uint8_t type);
```

Comment:
All these forward declarations are also excess. Just include required header.

Reply:
I've looked at the modern libraries, and this is a general style that I've picked from it. But there isn't matter , I might remove the declaration in.cpp

Reply:
As I can see, these functions don't use each other and the forward declarations or including the header seem unnecessary at all.

Reply:
Ok, I'll fix that.